services/spar/test-integration/Test/Spar/DataSpec.hs

```diff
@@ -203,20 +210,44 @@ testSPStoreID store unstore isalive = do
         isit <- runSparCassWithEnv $ isalive xid
         liftIO $ isit `shouldBe` True
 
-    context "after TTL" $ do
-      it "isAliveID returns False" $ do
-        xid :: SAML.ID a <- nextSAMLID
-        eol :: Time      <- addTime 2 <$> runSimpleSP getNow
-        () <- runSparCassWithEnv $ store xid eol
-        liftIO $ threadDelay 3000000
-        isit <- runSparCassWithEnv $ isalive xid
-        liftIO $ isit `shouldBe` False
 
-    context "after call to unstore" $ do
-      it "isAliveID returns False" $ do
-        xid :: SAML.ID a <- nextSAMLID
-        eol :: Time      <- addTime 5 <$> runSimpleSP getNow
-        () <- runSparCassWithEnv $ store xid eol
-        () <- runSparCassWithEnv $ unstore xid
-        isit <- runSparCassWithEnv $ isalive xid
-        liftIO $ isit `shouldBe` False
+-- | Test that when a team is deleted, all relevant data is pruned from the
+-- Spar database.
+testDeleteTeam :: SpecWith TestEnv
+testDeleteTeam = it "cleans up all the right tables after deletion" $ do
+    -- Create a team with two users and a SCIM token
+    (tok, (_uid, tid, idp)) <- registerIdPAndSCIMToken
+    user1 <- randomSCIMUser
+    user2 <- randomSCIMUser
+    storedUser1 <- createUser tok user1
+    storedUser2 <- createUser tok user2
+    -- Resolve the users' SSO ids
+    let getUid = read . unpack . SCIM.Common.id . SCIM.Meta.thing
+    ssoid1 <- getSsoidViaSelf (getUid storedUser1)
+    ssoid2 <- getSsoidViaSelf (getUid storedUser2)
+    -- Delete the team
+    runSparCass $ Data.deleteTeam tid
+    -- See that everything got cleaned up.
+    --
+    -- The token from 'team_provisioning_by_token':
+    do tokenInfo <- runSparCass $ Data.lookupScimToken tok
+       liftIO $ tokenInfo `shouldBe` Nothing
```

Comment:
[mumbling to myself] there is `shouldReturn`, which would make this more concise *and* arguably more readable, but it's in IO and not in MonadIO, so I'm not sure how to rewrite this.  i really should write mutants of all the `should*` functions in MonadIO...

Reply:
`should` functions should absolutely be in `MonadIO` and I've also been mumbling to myself about this for quite some time.

We can use http://hackage.haskell.org/package/hspec-expectations-lifted-0.10.0/docs/Test-Hspec-Expectations-Lifted.html but I'd rather have a homegrown library which would also contain some other common expectations that I end up reimplementing from time to time.